### ISSUE TITLE
php settings are misplaced on RHEL7

### ISSUE THREAD

**Issue by vchepkov** [mayflower/puppet-php#78]
I assume php::settings hiera should populate /etc/php.ini

```
php::settings:
  PHP/expose_php: Off
```

Instead it creates file /etc/php-cli.ini, which should be managed by php::cli::settings

I have PE 3.3 on RHEL7

**Comment by fpletz** [comment]
Thanks for reporting this. We definitely have a bug here for RedHat-based systems concering the location of the configuration files.

On a CentOS 7 VM:

```
$ php -v
PHP 5.4.16 (cli) (built: Oct 31 2014 12:59:36) 
Copyright (c) 1997-2013 The PHP Group
Zend Engine v2.4.0, Copyright (c) 1998-2013 Zend Technologies
```

```
$ php --ini
Configuration File (php.ini) Path: /etc
Loaded Configuration File:         /etc/php.ini
Scan for additional .ini files in: /etc/php.d
Additional .ini files parsed:      /etc/php.d/curl.ini,
/etc/php.d/fileinfo.ini,
/etc/php.d/json.ini,
/etc/php.d/phar.ini,
/etc/php.d/zip.ini
```

```
$ php-fpm -i
[..]
Configuration File (php.ini) Path => /etc
Loaded Configuration File => /etc/php.ini
Scan this dir for additional .ini files => /etc/php.d
Additional .ini files parsed => /etc/php.d/curl.ini,
/etc/php.d/fileinfo.ini,
/etc/php.d/json.ini,
/etc/php.d/phar.ini,
/etc/php.d/zip.ini
[..]
```

So both the cli and fpm share the same `php.ini` which is `/etc/php.ini`. Seems a bit weird coming from Debian/Ubuntu.

I think that on RedHat-based systems we should fail when trying to use `php::cli::settings` and `php::fpm::settings`, and set `php::settings` in `/etc/php.ini`. @globin Your opinion?

**Comment by globin** [comment]
Or maybe merge `php::cli::settings` and `php::fpm::settings` to `php::settings` and add a warning?

**Comment by cprior** [comment]
I am /very/ new to Puppet but my problem might be related:
- followed instructions at http://stdout.no/a-modern-puppet-master-from-scratch/
-  Puppetfile contains amongst other things:
classes:
  - php:
      ensure: latest
      php::settings:
        Date/date.timezone: Europe/Berlin
- the puppet agent runs without error (- mysql-server \    ensure running works as expected)
- grep -r Berlin /etc/php* returns no match

Ubuntu 14.04LTS, fresh install
php --ini
Configuration File (php.ini) Path: /etc/php5/cli
Loaded Configuration File:         /etc/php5/cli/php.ini
Scan for additional .ini files in: /etc/php5/cli/conf.d
Additional .ini files parsed:      /etc/php5/cli/conf.d/05-opcache.ini,
/etc/php5/cli/conf.d/10-pdo.ini,
/etc/php5/cli/conf.d/20-apcu.ini,
/etc/php5/cli/conf.d/20-gd.ini,
/etc/php5/cli/conf.d/20-json.ini,
/etc/php5/cli/conf.d/20-mysql.ini,
/etc/php5/cli/conf.d/20-mysqli.ini,
/etc/php5/cli/conf.d/20-pdo_mysql.ini,
/etc/php5/cli/conf.d/20-readline.ini

**Comment by fpletz** [comment]
@vchepkov Are you using the remi-php56 yum repo or the stock version of php? It seems that that version actually uses `/etc/php-cli.ini` for the cli settings.

Also, do have fpm disabled? Due to the php-fpm ini file being located at `/etc/php.ini`, it won't get written if you disabled fpm.

Because there isn't a global php configuration on every distribution, we merge the global settings from `php::settings` with those specific to fpm and cli and write the result to the respective ini files.

**Comment by vchepkov** [comment]
I am using this code:
```
  class { 'php':
    fpm        => true,
  }
```

hiera:
```
php::fpm::config::user: nginx
php::fpm::config::group: nginx
php::extentions:
  curl: {}
  gd: {}
```

**Comment by vchepkov** [comment]
Do you have any workaround ?

**Comment by minorOffense** [comment]
I think this is the right answer. Generate an ini file specific to fpm or cli and leave the global settings in php.ini

http://serverfault.com/questions/540051/is-there-an-equivalent-to-debians-cli-php-ini-on-redhat

**Comment by minorOffense** [comment]
I wrote a patch to fix the issue. But it needs some work. I'm not super familiar with Travis or rspec tests. If anyone can lend a hand in parsing the test results I can adjust the patch.

**Comment by fpletz** [comment]
@minorOffense Ah, thanks for looking into this! :cookie:

**Comment by fpletz** [comment]
Fixed by #132.
